plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentmodel/ContentTypeResource.java

```diff
@@ -0,0 +1,165 @@
+package org.entando.entando.plugins.jacms.web.contentmodel;
+
+import com.agiletec.aps.system.services.role.Permission;
+import com.agiletec.plugins.jacms.aps.system.services.contentmodel.model.*;
+import io.swagger.annotations.*;
+import org.entando.entando.aps.system.services.entity.model.*;
+import org.entando.entando.web.common.annotation.RestAccessControl;
+import org.entando.entando.web.common.model.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+public interface ContentTypeResource {
+
+
+    @ApiOperation(
+            value = "createContentType",
+            nickname = "createContentTypeUsingPOST",
+            response = ContentTypeDto.class,
+            tags={ "content-type-resource-controller", })
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = ContentTypeDto.class),
+            @ApiResponse(code = 201, message = "Created"),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden") })
+    @PostMapping("/plugins/cms/contentTypes")
+    ResponseEntity<ContentTypeDto> create(
+            @ApiParam(value = "contentType", required = true) @Valid @RequestBody ContentTypeDtoRequest contentType,
+            BindingResult bindingResult
+    ) throws URISyntaxException;
+
+
+    @ApiOperation(
+            value = "deleteContentType",
+            nickname = "deleteContentTypeUsingDELETE",
+            tags={ "content-type-resource-controller", })
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 204, message = "No Content"),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden") })
+    @DeleteMapping("/plugins/cms/contentTypes/{code}")
+    ResponseEntity<Void> delete(@ApiParam(value = "code", required = true) @PathVariable("code") String code);
+
+
+    @ApiOperation(
+            value = "getAllContentTypes",
+            nickname = "getAllContentTypesUsingGET",
+            response = ContentTypeDto.class,
+            responseContainer = "List",
+            tags={ "content-type-resource-controller", })
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = ContentTypeDto.class, responseContainer = "List"),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden") })
+    @GetMapping(path = "/plugins/cms/contentTypes")
+    @RestAccessControl(permission = Permission.SUPERUSER)
```

Comment:
Has the `@RestAccessControl` bug been fixed? I think we should use the `@RestMapping` if it is not fixed yet.

Reply:
The bug will be fixed in https://jira.entando.org/browse/EN-2277 
I'll push that one in before this story is finished

Reply:
This one is fixed now with the latest merge of 2277 in master and into EN-2162-2163